### PR TITLE
fix: gemini openai through signature compatibility

### PR DESCRIPTION
**Pull request by TejasGhatte**
## Summary

When a conversation switches from Gemini to OpenAI, Gemini embeds its `thoughtSignature` into tool call IDs using the format `<baseID>_ts_<signature>`. OpenAI rejects `call_id` values exceeding 64 characters, causing cross-provider conversation continuations to fail. This PR strips the embedded signature before sending IDs to OpenAI, while leaving the caller's input untouched so a subsequent Gemini turn can still recover the signature.

## Changes

- Introduced `ThoughtSignatureSeparator` (`_ts_`) as a shared constant in `core/providers/utils` and a `StripThoughtSignature` helper that returns the base ID by cutting at the separator.
- The Gemini `thoughtSignatureSeparator` constant now references the shared `ThoughtSignatureSeparator` instead of a local string literal.
- In the OpenAI chat converter (`ConvertBifrostMessagesToOpenAIMessages`), assistant tool call IDs and tool result `tool_call_id` values are stripped of any embedded signature. Structs are cloned before modification to avoid mutating the caller's input.
- In the OpenAI Responses converter (`ToOpenAIResponsesRequest`), `ResponsesToolMessage.CallID` is stripped the same way, also via a shallow clone.
- When a Gemini assistant message part has no `ThoughtSignature`, it is now assigned `skipThoughtSignatureValidator` to prevent downstream validation errors during cross-provider routing.
- Tests cover `StripThoughtSignature` edge cases, the chat converter stripping behavior (including immutability of the original input), and the Responses converter stripping behavior.

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Chore/CI

## Affected areas

- [x] Core (Go)
- [ ] Transports (HTTP)
- [x] Providers/Integrations
- [ ] Plugins
- [ ] UI (React)
- [ ] Docs

## How to test

```sh
go test ./core/providers/openai/... ./core/providers/gemini/... ./core/providers/utils/...
```

The new tests verify:
- `StripThoughtSignature` correctly handles IDs with and without embedded signatures.
- The OpenAI chat converter strips signatures from assistant tool call IDs and tool result IDs without mutating the original request.
- The OpenAI Responses converter strips signatures from `call_id` fields without mutating the original request.

## Breaking changes

- [x] No

## Security considerations

No auth, secrets, or PII implications. The stripping is deterministic and only affects tool call ID formatting on the wire.

## Checklist

- [ ] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [ ] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [ ] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by TejasGhatte** [comment]
* **#4842** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4842?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4810** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4810?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#4831** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4831?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4813** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/4813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: core/providers/utils/utils.go [hunk 1]
```diff
@@ -35,6 +35,22 @@ import (
...
+
+// StripThoughtSignature returns the base tool-call ID without any embedded provider
+// reasoning signature. It is deterministic, so a tool call and its matching output strip
+// to the same ID. Providers that cannot use the signature (e.g. OpenAI, which caps call_id
+// at 64 chars) call this before sending the ID upstream.
+func StripThoughtSignature(callID string) string {
+	if base, _, found := strings.Cut(callID, ThoughtSignatureSeparator); found {
+		return base
+	}
+	return callID
```
[the remark this answers is not in the document]
Comment (TejasGhatte) [reply core/providers/utils/utils.go:51]: can you check again

**Comment by Pratham-Mishra04** [comment]
### Merge activity

* **Jul 2, 1:44 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/4810).
* **Jul 2, 1:53 PM UTC**: [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/4810) rebased this pull request as part of a merge.
* **Jul 2, 1:54 PM UTC**: @Pratham-Mishra04 merged this pull request with [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/4810).
